src/node_jsvmapi.h

```diff
@@ -315,17 +315,18 @@ NODE_EXTERN napi_status napi_is_exception_pending(napi_env e, bool* result);
 NODE_EXTERN napi_status napi_get_and_clear_last_exception(napi_env e, napi_value* result);
 
 // Methods to provide node::Buffer functionality with napi types
-NODE_EXTERN napi_status napi_buffer_new(napi_env e,
-                                        char* data,
-                                        size_t size,
-                                        napi_value* result);
-NODE_EXTERN napi_status napi_buffer_copy(napi_env e,
-                                         const char* data,
-                                         size_t size,
-                                         napi_value* result);
-NODE_EXTERN napi_status napi_buffer_has_instance(napi_env e, napi_value v, bool* result);
-NODE_EXTERN napi_status napi_buffer_data(napi_env e, napi_value v, char** result);
-NODE_EXTERN napi_status napi_buffer_length(napi_env e, napi_value v, size_t* result);
+NODE_EXTERN napi_status napi_create_buffer(napi_env e,
```

Comment:
I'd actually prefer 2 separate APIs:
 * `napi_create_buffer` allocates and frees the buffer for you.
 * `napi_create_external_buffer` takes an external data pointer and finalize callback.

That would be consistent with the `napi_*_arraybuffer` APIs. Except `napi_create_external_arraybuffer` is missing a finalize callback. That was an oversight in my implementation. Do you want to add it, or should I add it separately?

JS `ArrayBuffer` and `node::Buffer` are almost the same thing anyway. As far as I know, the only reason node has its own buffer type is because the JS type didn't exist when Node got started.